src/deepsparse/transformers/pipelines/embedding_extraction.py

```diff
@@ -0,0 +1,422 @@
+        'deepsparse' and 'onnxruntime'. Default is 'deepsparse'
+    :param batch_size: static batch size to use for inference. Default is 1
+    :param num_cores: number of CPU cores to allocate for inference engine. None
+        specifies all available cores. Default is None
+    :param scheduler: (deepsparse only) kind of scheduler to execute with.
+        Pass None for the default
+    :param input_shapes: list of shapes to set ONNX the inputs to. Pass None
+        to use model as-is. Default is None
+    :param alias: optional name to give this pipeline instance, useful when
+        inferencing with multiple models. Default is None
+    :param sequence_length: sequence length to compile model and tokenizer for.
+        If a list of lengths is provided, then for each length, a model and
+        tokenizer will be compiled capable of handling that sequence length
+        (also known as a bucket). Default is 128
+    :param emb_extraction_layer: transformer layer number from which the embeddings
+        will be extracted. If None, leave the model unchanged. Default is -1
+        (last layer)
+    :param model_size: size of transformer model (size of hidden layer per token
+        if the model is cut). Default is 768
+    :param extraction_strategy: method of pooling embedding values. Currently
+        supported values are 'per_token', 'reduce_mean', 'reduce_max' and 'cls_token'.
+        Default is 'per_token'
+    :param show_progress_bar: token index(es) to extract from the embedding. Can
+        either be an integer representing the CLS token index, or a list of indexes
+        to extract. Default is 0
+    :param return_numpy: return embeddings a list of numpy arrays, list of lists
+        of floats otherwise. Default is True
+    :param context: context for engine. If None, then the engine will be initialized
+        with 2 streams to make use of parallel inference of labels. Default is None
+    """
+
+    def __init__(
+        self,
+        *,
+        emb_extraction_layer: Union[int, None] = -1,
+        model_size: int = 768,
+        extraction_strategy: str = "per_token",
+        show_progress_bar: bool = False,
+        return_numpy: bool = True,
+        context: Optional[Context] = None,
+        **kwargs,
+    ):
+        self._emb_extraction_layer = emb_extraction_layer
+        self._model_size = model_size
+        self._extraction_strategy = extraction_strategy
+        self._show_progress_bar = show_progress_bar
+        self._return_numpy = return_numpy
+
+        if context is None:
+            # num_streams is arbitrarily chosen to be any value >= 2
+            context = Context(num_cores=None, num_streams=2)
+
+        kwargs.update({"context": context})
+
+        self._thread_pool = ThreadPoolExecutor(
+            max_workers=context.num_streams or 2,
+            thread_name_prefix="deepsparse.pipelines.embedding_extraction",
+        )
```

Comment:
the multithreaded support here is not required functionality correct? if not, let's remove for now, and rely on upcoming support for multi-thread based dynamic batch

Reply:
Kind of. The Haystack's evaluation functions require that the model be capable capable of handling single queries (batch size 1) and ingesting lots of documents (it'd be nice if this took a reasonable amount of time). It doesn't break Haystack functionality as long as we keep batch size to 1.

Reply:
Decided to remove and enforce batch size to 1